tests/cxx/cxx17ns.cc

```diff
@@ -0,0 +1,33 @@
+//===--- cxx17ns.cc - test input file for iwyu ----------------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "tests/cxx/cxx17ns-i1.h"
+
+struct Two {
+  Two(a::b::c::One& one);
+  Two(a::b::One2& one);
+  Two(a::One3& one);
+};
+
+/**** IWYU_SUMMARY
+
+tests/cxx/cxx17ns.cc should add these lines:
+namespace a { struct One3; }
```

Comment:
Nice testcase! Is it possible to add coverage for the anonymous namespace cases too? I don't know if this would trigger that path:

    namespace { class Unused; }

but

    namespace a {
    namespace {
        struct Three {};
    }
    ...
    }

might.

Reply:
Oh, this is a very good point. So https://clang.llvm.org/extra/clang-tidy/checks/modernize-concat-nested-namespaces.html explicitly doesn't use that syntax for anonymous namespaces; I'll avoid the new syntax for that case as well.